### PR TITLE
Enable native AArch64 Ubuntu CI jobs

### PR DESCRIPTION
**Pull request by diegorusso**
The Arm runners are now enabled for the JIT builds.
We exclude the Arm runners on forks for both Tests and JIT jobs.

### CODE REVIEW

File: .github/workflows/jit.yml [hunk 1]
```diff
@@ -78,10 +78,10 @@ jobs:
             runner: macos-14
           - target: x86_64-unknown-linux-gnu/gcc
             architecture: x86_64
-            runner: ubuntu-22.04
+            runner: ubuntu-24.04
           - target: aarch64-unknown-linux-gnu/gcc
             architecture: aarch64
-            runner: ubuntu-22.04
+            runner: ${{ github.repository_owner == 'python' && 'ubuntu-24.04-aarch64' || 'ubuntu-24.04' }}
```
Comment (brandtbucher) [inline .github/workflows/jit.yml:85]: Maybe just add a comment here?

```suggestion
            # Forks don't have access to our paid AArch64 runners. These jobs are skipped below:
            runner: ${{ github.repository_owner == 'python' && 'ubuntu-24.04-aarch64' || 'ubuntu-24.04' }}
```

File: .github/workflows/jit.yml [hunk 2]
```diff
@@ -118,37 +118,18 @@ jobs:
           ./python.exe -m test --multiprocess 0 --timeout 4500 --verbose2 --verbose3
 
       - name: Native Linux
-        if: runner.os == 'Linux' && matrix.architecture == 'x86_64'
+        if: runner.os == 'Linux' && (matrix.architecture == 'x86_64' || github.repository_owner == 'python')
```
Comment (brandtbucher) [inline .github/workflows/jit.yml:123]:
```suggestion
        # Forks don't have access to our paid AArch64 runners. Skip those:
        if: runner.os == 'Linux' && (matrix.architecture == 'x86_64' || github.repository_owner == 'python')
```

**Comment by diegorusso** [comment]
This is the continuation of https://github.com/python/cpython/pull/125786

**Review by brandtbucher — APPROVED** [review]
JIT changes look good, thanks for tackling this! CI-driven development is no fun, I know.

**Review by hugovk — APPROVED** [review]

**Comment by hugovk** [comment]
Hmm, I do find it somewhat misleading that the jobs just "passes" immediately (well, not quite immediately, it takes about 11s to spin up and checkout), and would prefer to skip it entirely.

image

And when we do run, we're skipping three irrelevant jobs, and there's nothing much special we're sharing in the setup config to benefit from it (just checkout and setup-python):

image

And I did like the refactored one had more readable job names and grouped the three operating systems in the summary:

image

Compared to this:

image

But from a practical point of view it's not such a big deal, and the important thing is to start testing the new runners, so I've approved!

Thanks all for your work!

**Comment by brandtbucher** [comment]
> Hmm, I do find it somewhat misleading that the jobs just "passes" immediately (well, not quite immediately, it takes about 11s to spin up and checkout), and would prefer to skip it entirely.

But the alternative is just not running the job at all, right? Maybe when this happens we can just echo one of GitHub's magic warning messages to stdout so at least a yellow box appears?

I'm not strongly opposed to refactoring if many others feel that way, I just didn't think that this particular change warranted rewriting the whole file and adding a bunch of new logic. I personally find the current structure a easier to follow:

- All of the names and configurations of the jobs fit in one screenful.
- All of the commands that the jobs actually run, across all OSes, fit in one screenful.
- It's pretty easy to do things like support more than one LLVM version (we've done this in the past), etc.

If rewriting it so every OS is its own separate set of jobs would help in some real way, I'm probably okay with it in its own issue/PR. But simpler things like changing the names are easy enough to do in the current file. And it seems like we'll need "clever" tricks for handling the paid runners anyways.

**Review by savannahostrowski — APPROVED** [review]
Much simpler! Thanks Diego!

**Comment by diegorusso** [comment]
PR https://github.com/python/cpython/pull/125786 was back ported to 3.13. I don't think we need this for 3.13 as we don't build the JIT in this version. Thoughts?

**Pull request merged**